src/util.cpp

```diff
@@ -420,20 +417,29 @@ void ReleaseDirectoryLocks()
 }
 
 /** Interpret string as boolean, for argument parsing */
-static bool InterpretBool(const std::string& strValue)
+static bool InterpretBool(const std::string& val)
 {
-    if (strValue.empty())
-        return true;
-    return (atoi(strValue) != 0);
+    return val != "0";
```

Comment:
Looks like this will miss " 0" (whitespace), and "-0" (negative zero), which could happen as a typo on another argument.